ArduCopter/AP_Arming.cpp

```diff
@@ -611,7 +611,7 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
         const char *rc_item = "Throttle";
 #endif
         // check throttle is not too high - skips checks if arming from GCS/scripting in Guided,Guided_NoGPS or Auto 
-        if (!((AP_Arming::method_is_GCS(method) || method == AP_Arming::Method::SCRIPTING) && (copter.flightmode->mode_number() == Mode::Number::GUIDED || copter.flightmode->mode_number() == Mode::Number::GUIDED_NOGPS || copter.flightmode->mode_number() == Mode::Number::AUTO))) {
+        if (!((AP_Arming::method_is_GCS(method) || method == AP_Arming::Method::SCRIPTING) && copter.flightmode->allows_GCS_arming_with_throttle_high())) {
```

Comment:
Kind of misnamed here, given the scripting thing

Reply:
allows_GCS_or_SCR_arming_with_throttle_high ?

Reply:
> allows_GCS_or_SCR_arming_with_throttle_high ?

I have changed to this.